src/wallet/wallet.cpp

```diff
@@ -1970,6 +1970,32 @@ bool CWallet::SelectCoins(const CAmount& nTargetValue, set<pair<const CWalletTx*
     return res;
 }
 
+#if 0
+bool CWallet::SignTransaction(CMutableTransaction &tx)
+{
+    AssertLockHeld(cs_wallet); // mapWallet
+
+    // sign the new tx
+    CTransaction txNewConst(tx);
+    int nIn = 0;
+    for (const auto& input : tx.vin) {
+        std::map<uint256, CWalletTx>::const_iterator mi = mapWallet.find(input.prevout.hash);
+        if(mi == mapWallet.end() || input.prevout.n >= mi->second.tx->vout.size()) {
+            return false;
+        }
+        const CScript& scriptPubKey = mi->second.tx->vout[input.prevout.n].scriptPubKey;
+        const CAmount& amount = mi->second.tx->vout[input.prevout.n].nValue;
+        SignatureData sigdata;
+        if (!ProduceSignature(TransactionSignatureCreator(this, &txNewConst, nIn, amount, SIGHASH_ALL), scriptPubKey, sigdata)) {
+            return false;
+        }
+        UpdateTransaction(tx, nIn, sigdata);
+        nIn++;
+    }
+    return true;
+}
+#endif
+
```

Comment:
This added section doesn't seem to be needed or have any purpose.  Can we remove it?

Reply:
This looks to me like a very useful API -- basically the most fundamentally useful API for a wallet: sign a transaction.  Since we don't use it right now I commented it out.  However, I wanted to leave it in so we can use it later, and to preserve attribution

Reply:
no strong opinion only a mild preference to add it when needed rather than `ifdef` it to 0 

Other than that, AFAICT we should adapt the code to work also with the CASH chain, i.e. something like 
```
                 unsigned int sighashType = SIGHASH_ALL;
                if (chainActive.Tip()->IsforkActiveOnNextBlock(miningForkTime.value) && walletSignWithForkSig.value) sighashType |= SIGHASH_FORKID;
....
....
              if (!ProduceSignature(TransactionSignatureCreator(this, &txNewConst, nIn, amount, sighashType), scriptPubKey, sigdata)) {
```

or by passing `sighashType` to `CWallet::SignTransaction`